### ISSUE TITLE
Error parsing sampling strategy

### ISSUE THREAD

**Issue by mareq** [jaegertracing/jaeger-client-node#135]
This is probably problem of (configuration of) all-in-one docker image, so it is more request for help than an issue.

## TL;DR

Error: [`Error in parsing sampling strategy: SyntaxError: Unexpected token c in JSON at position 1`](https://github.com/uber/jaeger-client-node/blob/b830e0f60e71d01034ec091f92a6c6ac99f47eb5/src/samplers/remote_sampler.js#L136).

Question1: Can someone please point me to the solution of this problem? According to my current understanding, it would probably be some documentation explaining how to configure the remote sampler, which I missed.

Question2: I tried with const sampler (`sampler: { type: 'const', param: 1 }`), but that would result no traces at all appearing in Jaeger UI. Could someone please point me to some more detailed documentation for samplers and their configuration?

## Long Story

I start the all-in-one docker container with the command from [the official documentation](http://jaeger.readthedocs.io/en/latest/getting_started/):
```
$ docker run --rm -p5775:5775/udp -p6831:6831/udp -p6832:6832/udp -p5778:5778 -p16686:16686 -p14268:14268 jaegertracing/all-in-one:latest
{"level":"info","ts":1500067556.8956976,"caller":"tchannel/bulider.go:95","msg":"Enabling service discovery","service":"jaeger-collector"}
{"level":"info","ts":1500067556.895823,"caller":"peerlistmgr/peer_list_mgr.go:117","msg":"Registering active peer","peer":"127.0.0.1:14267"}
{"level":"info","ts":1500067556.8969154,"caller":"standalone/main.go:82","msg":"Starting agent"}
{"level":"info","ts":1500067556.8984168,"caller":"standalone/main.go:117","msg":"Starting jaeger-collector TChannel server","port":14267}
{"level":"info","ts":1500067556.898527,"caller":"standalone/main.go:124","msg":"Starting jaeger-collector HTTP server","http-port":14268}
{"level":"info","ts":1500067556.9013727,"caller":"standalone/main.go:174","msg":"Starting jaeger-query HTTP server","port":16686}
{"level":"info","ts":1500067557.896384,"caller":"peerlistmgr/peer_list_mgr.go:165","msg":"Not enough connected peers","connected":0,"required":1}
{"level":"info","ts":1500067557.896504,"caller":"peerlistmgr/peer_list_mgr.go:172","msg":"Trying to connect to peer","host:port":"127.0.0.1:14267"}
{"level":"info","ts":1500067557.8978744,"caller":"peerlistmgr/peer_list_mgr.go:177","msg":"Connected to peer","host:port":"[::]:14267"}
```

Then I start the sample service with almost default configuration passed to the tracer:
```
const config = {
  'serviceName': 'foo'
};
const options = {
  'logger': console
};
const tracer = initTracer(config, options);
```

After sending some requests, it looks it is working (I see some traces in the Jaeger UI), but I keep getting the JSON parsing error while the tracer attempts to retrieve sampling strategy:

```
$ npm run start
Initializing Jaeger Tracer with RemoteReporter and RemoteSampler
API initialized
Error in parsing sampling strategy: SyntaxError: Unexpected token c in JSON at position 1.
```

This is due to the following problem:
```
$ curl -sS -i -H'Content-Type: application/json' --connect-timeout 10 --max-time 60 -X GET 'http://localhost:5778/sampling?service=foo'
HTTP/1.1 500 Internal Server Error
Content-Type: text/plain; charset=utf-8
X-Content-Type-Options: nosniff
Date: Mon, 17 Jul 2017 14:33:40 GMT
Content-Length: 37

tcollector error: no peers available
```

**Comment by black-adder** [comment]
Hi there

Question1: this error currently occurs since we haven't open sourced the sampling processor logic in the collector yet. It's on our TODO list. I'd advise using a different sampling strategy.

Question2: this is weird, you should be seeing 100% sampling using the const sampler. Can you post the config and the options you pass into the tracer? Also you can add this to the config:
```
const config = {
  'reporter': {
    'logSpans': true
  }
};
```
The above will log all the traces that are being sampled. This can help us narrow down where the issue is. I'll add some documentation on the different sampling strategies.

**Comment by mareq** [comment]
Hi, thank you for fast response!

Please disregard my Question2, it was couple of silly problems between my keyboard and chair, now resolved.

As for the Question1, what would be your recommendation for our approach to the sampling strategy? Should we wait for the sampling strategy logic to be open sourced (what kind of timeline do you have in mind?) and use e.g. ProbabilisticSampler in the meantime? Regarding the documentation on the sampling strategies, if you are open to sharing some tips on it, we may be able to help with the documentation.

**Comment by black-adder** [comment]
I'd recommend using ProbabilisticSampler or the Const sampler in the meantime depending on the QPS of your service.

We support 4(5) samplers:
1) Const - depending on the param, this sampler samples everything or nothing (no in between)
2) Probabilistic - probability based, the default is 0.001 (1 in 1000). ie one out of every 1000 requests will be traced.
3) Ratelimiting - Takes in param which stands for `maxTracesPerSecond`. ie if param is 2, every second the tracer will trace AT MOST 2 requests.
4) Remote - This strategy fetches sampling strategies from the collector once a minute (this interval is configurable). This returned strategy can either be probabilistic or adaptive.
5) Adaptive - A sampling strategy that is used by Remote. This strategy dynamically adjusts the sampling probability such that each service/endpoint is traced at a specific rate.

4 and 5 are still in the process of being open sourced. We have an OKR to get them done this quarter.

**Comment by yurishkuro** [comment]
@mareq it looks like your questions have been answers, please re-open if not and clarify the issue.
